crates/ruff_cli/src/commands/format.rs

```diff
@@ -347,6 +399,23 @@ impl Display for FormatSummary<'_> {
                     formatted += 1;
                 }
                 FormatResult::Unchanged => unchanged += 1,
+                FormatResult::Diff {
+                    path,
+                    unformatted,
+                    formatted: formatted_source,
+                } => {
+                    // Diff
+                    let mut writer = stdout().lock();
+                    let text_diff = TextDiff::from_lines(
+                        unformatted.source_code(),
+                        formatted_source.source_code(),
+                    );
+                    let mut unified_diff = text_diff.unified_diff();
+                    unified_diff.header(&fs::relativize_path(path), &fs::relativize_path(path));
+                    unified_diff.to_writer(&mut writer).unwrap();
+
+                    formatted += 1;
+                }
```

Comment:
Nit: It feels a bit strange that the diff is written as part of the `FormatSummary`'s `Display` implementation (the diff isn't part of the summary). 

I would prefer moving the diff printing out of the summary implementation and handle it explicitly when the `FormatMode` is `Diff` (also allows to locking `stdout` exactly once)

Reply:
I went with renaming `FormatSummary` to `FormatResults` and giving it two different write methods. We unfortunately lose the fmt impl but we get proper separation